Idrall/Check.idr

```diff
@@ -90,14 +97,26 @@ mutual
   aEquivMaybe _ _ _ _ _ = False
 
   aEquivList : (i : Integer) ->
-                Namespace -> List (Expr Void) ->
-                Namespace -> List (Expr Void) -> Bool
+               Namespace -> List (Expr Void) ->
+               Namespace -> List (Expr Void) -> Bool
   aEquivList i ns1 [] ns2 [] = True
   aEquivList i ns1 (x :: xs) ns2 (y :: ys) =
     aEquivHelper i ns1 x ns2 y &&
     aEquivList i ns1 xs ns2 ys
   aEquivList i ns1 _ ns2 _ = False
 
+  aEquivUnion : (i : Integer) ->
+                Namespace -> List (FieldName, (Maybe (Expr Void))) ->
+                Namespace -> List (FieldName, (Maybe (Expr Void))) -> Bool
+  aEquivUnion i ns1 [] ns2 [] = True
+  aEquivUnion i ns1 ((k, Nothing) :: xs) ns2 ((k', Nothing) :: ys) =
+    k == k' && aEquivUnion i ns1 xs ns2 ys
+  aEquivUnion i ns1 ((k, Just v) :: xs) ns2 ((k', Just v') :: ys) =
+    k == k' &&
+    aEquivHelper i ns1 v ns2 v' &&
+    aEquivUnion i ns1 xs ns2 ys
+  aEquivUnion i ns1 _ ns2 _ = False
```

Comment:
Hmm, so I'm not sure how dhall works, but we might have to be more careful here. Is `<Foo: Int, Bar: String>` the same type as `<Bar: String, Foo: Int>`. Some languages with these kind of types allow the order to be different, but still be "same" type.

Reply:
It's a `SortedMap` and keys should be unique (the check for which i haven't implemented yet now that I'm thinking about it), so I think the should be enough to check for alpha equivalence.